### ISSUE TITLE
Testing Results

### ISSUE THREAD

**Issue by moovida** [mikes222/mapsforge_flutter#2]
Hi, I made some tests with mapsforge_flutter. I hope you are ok with me sharing here the results found.

1) In my tests with the example up, I found that on lower zoomlevels, the app is quite slower than mapsforge and freezes the rendering cycle. Instead at higher zoomlevels, the performance is quite similar and with caching definitely acceptable. 

2) I then tried to use mapsforge_flutter as a tile provider for the flutter_map project. This is interesting because flutter_map already has everything necessary to do markers, lines and polygons, as well as online tiling services and mbtiles. It also has a nice clustering option. 

The result looks quite nice to me:

![image](https://user-images.githubusercontent.com/390250/60010621-6f5b3100-9678-11e9-9e2b-86ea7ee197df.png)

The issues I found are the following:

* same as for your example app, when the zoomlevels are low, the tile generation get's very slow
* one second thing, and maybe a bit more problematic, is the multiplication of the labels on the map on certain lines. Having no real idea about mapsforge's internals, I am not sure what that might be. AN image might be more than many words:

In this image the label of the river "Adige-Po" is repeated many times:

![image](https://user-images.githubusercontent.com/390250/60011182-83ebf900-9679-11e9-8c0f-30a602adf05f.png)

One last thing just to make sure. I tried to change the tilesize, which broke it. I assume tilesize at the moment should work only with 256, right?

**Comment by mikes222** [comment]
ad 1: This comes because flutter runs in a single thread. When reading large portions of the mapfile we have major performance issues. A have already opened an issue at the flutter team but since this is a design constraint of flutter we may be in trouble...

ad 2: Thank you for the test. Looks good. Online tiles are now supported by this project too. 
The labels not rotated is a known issue. We need to rotate the label, calculate the size of the text and skip the next label if it intersects with the current one. I think this should be done in the `WayTextContainer`. Take a look at the original project. 

The tilesize of the online tiles are also 256 pixels. We would need to investigate but this has a very low priority for me.

**Comment by mikes222** [comment]
way-labels are now rotated and skipped if too close together. You could retry with newest version.

**Comment by moovida** [comment]
Hi @mikes222, great news, with way-labels following the lines this looks much better! 

I have tried to tweak a little bit the textscale in order to have the labels readable on high density displays.

I tried changing the userscale: 
```
_displayModel.setUserScaleFactor(2.5);
```
it changes the scale of the drawings but not the labels:

![image](https://user-images.githubusercontent.com/390250/61167667-6906f900-a542-11e9-8238-c43b28bc447d.png)

I also tried to change the textsize of the render job:
```
RendererJob mapGeneratorJob = new RendererJob(tile, _multiMapDataStore,
        _renderTheme, _displayModel, userScaleFactor, false, false);
```
but without effect.

I am still using mapsforge_flutter as the tiles generator inside flutter_map, so that might be part of teh problem.

Thanks for you amazing work!
